pkg/server/service/loadbalancer/wrr/wrr.go

```diff
@@ -151,5 +149,15 @@ func (b *Balancer) AddService(name string, handler http.Handler, weight *int) {
 	if weight != nil {
 		w = *weight
 	}
-	b.handlers = append(b.handlers, &namedHandler{Handler: handler, name: name, weight: w})
+	if w <= 0 { // non-positive weight is meaningless
+		return
+	}
+	h := &namedHandler{Handler: handler, name: name, weight: float64(w)}
+	h.deadline = b.curDeadline + 1/h.weight
+	if h.deadline < b.baseDeadline {
+		h.deadline = math.Ceil((b.baseDeadline-h.deadline)*h.weight) / h.weight
+	}
```

Comment:
Could you explain this part please? I understand neither its rationale, nor its intended effect.

Reply:
I'd like to remove this.

---

It has effects.
Suppose we have 1000 hosts.
900 of them have weight 100, as a1, a2, a3, ..., a900
100 of them have weight 1, as b1, b2, b3,...,b100
so, in the picking loop, we always pick ax first.
We need to pick a about 90000 times before we pick the first b.
Sometimes, when ServeHTTP is called 50000 times, endpoints refreshed, so we pick a1, a2, a3 again, as b is never picked.

the Ceil is used for a random pick in the cycle. it makes b has the opportunity to be picked first.

Reply:
Removed